src/rust/scheduler/scheduler.rs

```diff
@@ -199,13 +198,20 @@ impl Scheduler {
             }
             // There is some dropped task in this page, so iterate through it.
             if dropped != 0 {
+                let mut task_ids: RefMut<HashMap<u64, usize>> = self.task_ids.borrow_mut();
+
                 // Handle dropped tasks only.
                 for subpage_ix in BitIter::from(dropped) {
-                    if subpage_ix != 0 {
-                        let ix: usize = (page_ix << WAKER_BIT_LENGTH_SHIFT) + subpage_ix;
-                        tasks.remove(ix);
-                        pages[page_ix].clear(subpage_ix);
-                    }
+                    let index: usize = (page_ix << WAKER_BIT_LENGTH_SHIFT) + subpage_ix;
+                    println!("dropping task {:?}", index);
+                    let len: usize = task_ids.len();
+                    // Remove the task id that points to this offset.
+                    // We could build a backmap, but we do not expect drop to happen very often.
+                    task_ids.retain(|_, v| *v != index);
+                    // If there is more than one task id pointing at the offset, something has gone very wrong.
+                    assert!(task_ids.len() == len - 1);
```

Comment:
Could you add a message to this `assert!()` statement? If we reach this bad condition, we can debug without having to open the source code and reading out the comment above.